nightfall-optimist/src/routes/proposer.mjs

```diff
@@ -256,7 +268,9 @@ router.post('/payment', async (req, res, next) => {
  * for a parameterless function - therefore it's a constant and could be pre-
  * computed by the app that calls this endpoint.
  */
-router.get('/change', async (req, res, next) => {
+router.get('/change', auth, async (req, res, next) => {
```

Comment:
I think this function doesn’t need to be protected. Any one can call changeProposer

Reply:
I thought about that. On the other hand, I wonder if there's any situation where the proposer may not want to call changeProposer, for example to save on gas costs. I'm happy to open this route to the public though, if it makes more sense to you

Reply:
@daveroga please, comment on this

Reply:
So I just submitted a new commit. As we're going for a 1:1 implementation, I simply opened the three routes that are currently being used by the proposer app:

```
/offchain-transaction
/mempool
/:contract
```

Everything else should have the `auth` middleware, for consistency's sake. I'm happy to open other routes if @daveroga thinks it's worth it, some of them make perfect sense to be open (like `getCurrentProposer` and etc)

Reply:
`/fee` also was added for the block explorer and should be available for the user to know the fees

Reply:
`changeCurrentProposer` could be called by anyone, although the interested one is the next proposer. There is no issue in someone to call this. If the current proposer is still in the challenge period it will revert and if not it will change to the next proposer.